### PR TITLE
[Task-3][Chore] Configure view_component and view_component_storybook

### LINKED ISSUE #3: [Task-3][Chore] Configure view_component and view_component_storybook
## Why

We need to configure view_component and view_component_storybook for our next features.
 
## Acceptance Criteria

- Install the gem [view_component](https://viewcomponent.org/guide/getting-started.html) & [view_component_storybook](https://github.com/jonspalmer/view_component_storybook)
- Configure `view_component_storybook` for deployment
- Deploy a story to prove that it is running smoothly

## Resources

- Blog to deploy storybook - https://orbit.love/blog/building-a-component-library-in-rails-with-storybook
- Simple example - https://jonspalmer.github.io/view_component_storybook/guide/getting-started.html#quick-start

### PR DESCRIPTION
**Pull request by malparty**
- #3 

## What happened 👀

- Configure View Component 
- Create a Button ViewComponent (yeah, not original but easy!)
- Configure Story Book
- Add ReadMe get started doc (same as 'Insights' below)

### Configure View Component 

- Add `PostsController` with the basic `#index` method to have a root page

### Configure Story Book

> ℹ️ Apply the below configuration **once** set up your local machine with StoryBook

- Setup Storybook server locally
  ```shell
  yarn add @storybook/server @storybook/addon-controls --dev
  ```

### Push a new Story in StoryBook

> ℹ️ Apply the below commands **each time** you want to push a new component into the StoryBook
> _We assume you already built your component and its stories_

- Convert each ruby story into a JSON file:
  ```shell
  rake view_component_storybook:write_stories_json 
  ```

- Build the StoryBook
  ```shell
  yarn storybook:build
  ```

> The StoryBook is a static website located inside the `/public/_storybook` so it will be deployed **with** the rails application.

### View Component Preview:

| ViewComponent Preview | Root page rendering the component|
|---|---|
|StoryBookFirst_preview|image|

### StoryBook

StoryBookFirst

### CODE REVIEW

File: test/components/stories/button_component.stories.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    {
+      "name": "link",
+      "parameters": {
+        "server": {
+          "id": "button_component/link"
+        }
+      }
+    }
+  ]
+}
```
Comment (rafayet-monon) [inline test/components/stories/button_component.stories.json]: Let's add a new line.

**Comment by rafayet-monon** [comment]
Can we do something about the failed tests? @malparty

File: test/components/stories/button_component.stories.json (hunk 1 above)
Comment (malparty) [reply test/components/stories/button_component.stories.json]: This file is auto-generated... Should we manually add a new line (there will be more and more files re-generated each time) 🤔

**Comment by malparty** [comment]
> Can we do something about the failed tests? @malparty

Yes, same as I did in my Ruby Lab: Remove the check for test coverage... as it will require a lot of work to fix this while keeping the test coverage check.
Ok for you @rafayet-monon ?

**Comment by junan** [comment]
@malparty Can we deploy it to staging and check if it is working there?

**Comment by rafayet-monon** [comment]
> Yes, same as I did in my Ruby Lab: Remove the check for test coverage... as it will require a lot of work to fix this while keeping the test coverage check.
> Ok for you @rafayet-monon ?

Should we use Rspec for testing? https://viewcomponent.org/guide/testing.html#rspec-configuration 

As we use rspec for everything I think that should be ideal. And test files in `test/` folder is causing this CI break. What do you think?

**Comment by malparty** [comment]
> Can we do something about the failed tests? @malparty

For rubocop, as component stories will often have empty blocks, I've added an exclude. 
✅ Done in 1c69f20 ;-)

**Comment by malparty** [comment]
> > Yes, same as I did in my Ruby Lab: Remove the check for test coverage... as it will require a lot of work to fix this while keeping the test coverage check.
> > Ok for you @rafayet-monon ?
> 
> Should we use Rspec for testing? https://viewcomponent.org/guide/testing.html#rspec-configuration
> 
> As we use rspec for everything I think that should be ideal. And test files in `test/` folder is causing this CI break. What do you think?

Config and example provided in here: 1c5d7ce 🚀

**Comment by junan** [comment]
@malparty The test is failing now(`/spec/codebase/codebase_spec.rb`), Maybe we can just comment on it out to ignore them and make the test pass(As test pass is required to deploy)

And the docker building is failed.  I am not fully sure what went wrong. It it seems
 `require "view_component/storybook/engine"` generating the error while docker execute `docker-assets-precompile`
 Can you please check it a bit?
 
Screen Shot 2021-09-21 at 11 19 25 AM

**Review by rafayet-monon — APPROVED** [review]

**Review by junan — APPROVED** [review]

**Pull request merged**